pulse2percept/implants.py

```diff
@@ -327,11 +398,8 @@ def __init__(self, etype, radii, xs, ys, hs=0, names=None):
             # If not every electrode has a name, replace with None's
             names = np.array([None] * radii.size)
 
-        self.etype = etype
-        self.num_electrodes = names.size
-        self.electrodes = []
         for r, x, y, h, n in zip(radii, xs, ys, hs, names):
-            self.electrodes.append(Electrode(etype, r, x, y, h, n))
+            self.add_electrode(Electrode(self.etype, r, x, y, h, n))
```

Comment:
Random aside: right now, access to individual electrodes is done either by a name, or by an index. In both cases, ultimately the access is mediated by the index (`__getitem__` calls to `get_index` etc.). It occurs to me that instead of a list, the electrodes could be stored in a dict, with either integer keys (the case where no names are provided) or the names as keys. Might simplify some of the code below (?). Would it have any unwanted consequences? We could still iterate over the dict items (in `__iter__` below).